### ISSUE TITLE
Evaluation or comparison with alternatives (particularly Workrave) could be more efficient

### ISSUE THREAD

**Issue by Chealer** [hovancik/stretchly#1019]
I was recently informed that an important issue in Workrave could be avoided by using Stretchly instead, which got me to evaluate Stretchly to see if I should replace Workrave with it. As a longtime Workrave user, I was surprised I hadn't heard about Stretchly before it got so mature.

In the end, it took me many minutes to confirm that Stretchly still didn't match Workrave for my needs. As Stretchly and Workrave both address pretty much the same platforms and the same potential users, many more people are likely to "waste" some time in that process. As Stretchly's future is visibly more promising than Workrave's, comparison is likely to only get more difficult unless it is somehow facilitated.

There are 2 possible approaches:

1. Facilitating Stretchly evaluation
2. Providing a comparison with alternatives

I believe the best way to achieve the first approach would be to sort Stretchly issues by importance by default, or at least ensure its ITS allows to order tickets by decreasing importance.

The second approach would be something for Stretchly's website. I suppose it would be best to add a "Compare" page dedicated to that. A basic implementation could be to just link to an issue report which would allow everyone to provide input, like this one. Or a link to a comparison of multiple RSI management utilities, if a fair one can be found. Or, it could be 2 columns, one with each product's advantages.

If Stretchly offers its own comparison with Workrave, it could at least mention the following advantages (updated in February 2025 for Workrave 1.10.48 vs Stretchly 1.17.2):

1. [Possibility to suspend/pause breaks for a limited time](https://github.com/rcaelers/workrave/issues/98)
2. Synchronization with Microsoft Windows Do Not Disturb mode, [only supported starting from Workrave 1.11](https://github.com/rcaelers/workrave/issues/226), although unfortunately limited as reported in #1313 

...as well as the following main disadvantages:

1. Sounds:
    1. [The sound when a mini break ends is the same as the one when a long break ends](https://github.com/hovancik/stretchly/issues/687) by default, and the GUI does not allow changing that. While the sound is OK for long breaks, [it is too loud/disturbing for mini breaks](https://github.com/hovancik/stretchly/issues/1564).
    2. [The sound proposing a break is the same as the sound of any system notification](https://github.com/hovancik/stretchly/issues/1565), for example an incoming email.
2. The break window is unnecessarily very large by default (this can be adjusted, but [not easily](https://github.com/hovancik/stretchly/issues/1561)).
3. [Idleness detection is poor, resulting in numerous untimely break proposals](https://github.com/hovancik/stretchly/issues/1560).
4. [There are no usage statistics and there is no history of breaks.](https://github.com/hovancik/stretchly/issues/266)
5. Bloat
    1. installed size (see 2025 comments in ticket #1194 for instance). Stretchly is about 7 times heavier (350 MB as of 1.17.2 on Windows x64, vs 50 MB for Workrave 1.10.48).
    2. download size (approximately 5 times bigger)
    3. RAM consumption (as of 2025-02 on Windows x64, roughly between 107 and 249 MB (see ticket #1579) for Stretchly 1.17.2 vs 11 to 33 MB for Workrave 1.10)

In terms of foundations (development perspectives), Stretchly is implemented in JavaScript rather than C++, which may be more appropriate. On the other hand, it has no discussion forums (like the workrave-user mailing list).

**Comment by Chealer** [comment]
[The README file's _Known issues_ section](https://github.com/hovancik/stretchly/blob/trunk/README.md#known-issues) already takes the first approach, but unfortunately fails to mention several important issues.

**Comment by hovancik** [comment]
Well, *Known issues* mention only technical issues (that I am not able to fix) not the *usage issues*: Each person likes something else so *issue* for one use is good feature for someone else. 

With comparisons, there are many alternatives, not only Workrawe, and to be honest, I am not interested in doing comparisons as I am not checking the competitors at all. For example, AlternativeTo lists over 40 alternatives: https://alternativeto.net/software/stretchly/. Then it becomes: which apps to compare to? All 40? What app should do the comparison? One of them? All of them? What and how to compare? 

 I guess it's on the user in the end to do the research and  figure out what they like, as different people like different things.

**Comment by Chealer** [comment]
> Well, _Known issues_ mention only technical issues (that I am not able to fix) not the _usage issues_: 

If you want to keep a specific list of issues which you are not able to fix, this can be done by having an "Issues in need of technical help" section, which could be a subsection of _Known issues_.

That being said, as this information is likely not relevant to users, the vast majority of projects rather achieve this using their ITS. A "Help needed" tag can be created and applied to tickets about the issues in question.

> Each person likes something else so _issue_ for one use is good feature for someone else.

A list of issues in a program is generally not considered as a list of issues affecting every user, but if you feel that distinction is important, it is perfectly possible to categorize issues, for example in a list of _Issues for all users_ and another list of _Issues for specific users_.

> With comparisons, there are many alternatives, not only Workrawe, and to be honest, I am not interested in doing comparisons as I am not checking the competitors at all.

Just to be clear, unless there is a mention to the contrary, a ticket is not a request for anyone in particular to do a change. It is a report about an issue which can (if we ignore permissions) be tackled by any contributor.


> For example, AlternativeTo lists over 40 alternatives: https://alternativeto.net/software/stretchly/. Then it becomes: which apps to compare to? All 40? What app should do the comparison? One of them? All of them? What and how to compare?
> 
> I guess it's on the user in the end to do the research and figure out what they like, as different people like different things.

The user will always have to decide what they ultimately want to use. This ticket is about _facilitating_ comparison, and minimizing the research and time users will have to spend in that process. You ask good questions, and I don't have answers as good. If there was a fair comparison between all these alternatives, linking to it would surely be best, but I am not aware of any existing one. However, from all these listed alternatives, RSIBreak is (unless that has changed) specific to GNU/Linux, Time Out is reportedly specific to macOS, and Eyes Relax, Eyeleo, Eye Defender and Iris all have a purpose different from Stretchly (judging by their name). The rest don't even have the tenth of Stretchly's popularity (judging by the "likes" recorded on that site). So Workrave is _by far_ Stretchly's main competitor.

That being said, I shall remind that this ticket is asking to facilitate comparison _or evaluation_. If you find picking a number of alternatives to compare with too difficult, let's start with facilitating evaluation.

**Comment by Chealer** [comment]
Is there a reason why this ticket is closed?

**Comment by hovancik** [comment]
Because I don't plan to maintain any comparison with similar apps section

**Comment by Chealer** [comment]
@hovancik I see that you may be Stretchly's main contributor, but this does not prevent others from contributing to Stretchly. In fact, someone else may become its main contributor soon. As mentioned previously, this is not a request for you (or whoever would be Stretchly's main developer), nor even any of Stretchly's past developers, to solve the issue.

In fact, this ticket is not even a request at all; it is an _issue report_. As the issue persists, there is an implicit request to solve it, but the fact that a given developer has no plan to do so unfortunately does not constitute a solution to the issue.

That being said, at the risk of repeating myself, I shall use this opportunity to remind that this issue can be solved by facilitating comparison **or evaluation**. If you find comparing too difficult, let's start with facilitating evaluation.

**Comment by wolftune** [comment]
FWIW, I just did some comparison, and I appreciate the value of this, but indeed *maintaining* comparisons with projects all being moving targets is really cumbersome.

If anyone really works on this, it should IMO prioritize Free/Libre/Open options. That brings down the dozens of programs to just a few: Stretchly, SafeEyes, WorkRave, RSI Break, BreakTimer. And of these, I think really only Stretchly, SafeEyes, and WorkRave are competitive at this time.

**Comment by Chealer** [comment]
> FWIW, I just did some comparison, and I appreciate the value of this, but indeed _maintaining_ comparisons with projects all being moving targets is really cumbersome.
 
Good point. So if a comparison table is created, instead of having pure boolean values like "No", the values would better be versioned (at the very least in the code, but probably best to display to readers too). For example, "No (2.1)"

**Comment by Chealer** [comment]
I reevaluated Stretchly and updated the above comparison for 1.17.2. Although I added a few issues, these are not necessarily new, and there is no doubt that even though Workrave evolved a bit since my first evaluation, Stretchly evolved a lot more and largely closed the gap. The comparison is far from exhaustive, but I think it represents a reasonable roadmap of what's missing to catch up.

I will stay with Workrave for now, but I'd be happy to reevaluate in a couple years and struggle to establish which is best!

**Comment by wolftune** [comment]
FWIW, I have switched to the newest option which I think is all-around best: https://github.com/AllanChain/sane-break/ it uses the same style as WorkRave but is much newer and updated, efficient codebase rather than Electron etc. There are some open issue tickets there regarding getting it to meet all the functions of WorkRave, and many have already been done.

**Comment by herrbrixter** [comment]
> Each person likes something else so _issue_ for one use is good feature for someone else.

This is true. I love Stretchly's simplicity.

Some of the "issues" @Chealer  listed are "non-issues" to me. Some of the "disadvantages" listed are Stretchly's "advantages".

> * Sounds:

Small matter - I just need Stretchly to beep.

> * The break window is unnecessarily very large by default (this can be adjusted, but [not easily](https://github.com/hovancik/stretchly/issues/1561)).

Advantage - That is exactly what I like about Stretchly, a giant obstrusive banner that forces me to take a break.

> * [Idleness detection is poor, resulting in numerous untimely break proposals](https://github.com/hovancik/stretchly/issues/1560).

Small matter - I want Stretchly to tell me when to have a break, and not tell me when NOT to have a break. As you said, idle time detection is hard. If I were to err to one side, I would rather have Stretchly think I'm active when I'm idle, rather than Stretchly think that I'm idle when I'm active.

> * [There are no usage statistics and there is no history of breaks.](https://github.com/hovancik/stretchly/issues/266)

Advantage - I don't need this. Don't make it too complicated.

> * Bloat

No issue - Being a bit larger or taking more memory is not an issue. I have enough hard drive space an RAM.


In summary, @Chealer 's complaints are just arguing over small details. There is no need for Stretchly to close the gap with WorkRave. Many people like Stretchly for the exact reason that it is NOT WorkRave. Stop trying to make Stretchly a WorkRave-clone. If you like WorkRave better, then use WorkRave.

**Comment by Chealer** [comment]
> Some of the "issues" [@Chealer](https://github.com/Chealer) listed are "non-issues" to me. Some of the "disadvantages" listed are Strecthly's "advantages".
> 
> > * Sounds:
> 
> Small matter - I just need Stretchly to beep.

You're not the only intended audience though. Some of us have co-occupants and think about others.

 
> > * The break window is unnecessarily very large by default (this can be adjusted, but [not easily](https://github.com/hovancik/stretchly/issues/1561)).
> 
> Advantage - That is exactly what I like about Stretchly, a giant obstrusive banner that forces me to take a break.

If you're going to discuss Workrave, please evaluate it properly. Workrave can pamper you even more than Stretchly, it just doesn't doesn't treat its default user as a teenager.

 
> > * [Idleness detection is poor, resulting in numerous untimely break proposals](https://github.com/hovancik/stretchly/issues/1560).
> 
> Small matter - I want Stretchly to tell me when to have a break, and not tell me when NOT to have a break. As you said, idle time detection is hard. If I were to err to one side, I would rather have Stretchly think I'm active when I'm idle, rather than Stretchly think that I'm idle when I'm active.

I suggest you give Stretchly a real try or read the reports before trying to contribute. What I wrote is that _distinguishing activity from idleness is far from trivial, so perfection is certainly not possible_. This doesn't make it acceptable to initiate a break when the only activity which happened in 10 hours is a press on the power button. This matter alone would make permanently switching to Stretchly inconceivable in its current state, at least for me.


> > * [There are no usage statistics and there is no history of breaks.](https://github.com/hovancik/stretchly/issues/266)
> 
> Advantage - I don't need this. Don't make it too complicated.

Nor do you need to use it―it's just a screen you can bring up when you need it.


> > * Bloat
> 
> No issue - Being a bit larger or taking more memory is not an issue. I have enough hard drive space an RAM.

Taking memory is certainly an issue, and 3 to 10 times more is not what I'd call "a bit" more. Hard drives are getting rarer. If I didn't have enough space, I wouldn't have evaluated Stretchly is such detail, but I can assure you my own (high-end) main PC has been at times stretched in both permanent storage and RAM, and our systems are not representative of the entire population. Whether or not it reduces accessibility, greater size affects performance, at least at install time.
 
> In summary, a lot of [@Chealer](https://github.com/Chealer) 's complaints are just arguing over small details. There is no need for Stretchly to close the gap with WorkRave. Many people like Stretchly for the exact reason that it is NOT WorkRave. Stop trying to make Stretchly a WorkRave clone.

Please read. There are no complaints/arguments above, nor any request to close the gap with Workrave. The issue this tracks is the excessive effort needed to evaluate Stretchly or compare it with alternatives.

If you have a gripe with Workrave, please be constructive and start by ensuring your issues are tracked. If you think Stretchly's purpose/audience differs from Workrave's, please indicate how, so that this can be documented. If you're unsatisfied with the above comparison, at least propose yours, or probably even better: help us migrate to a proper ITS which can rank issues by importance.

**Comment by hovancik** [comment]
Stretchly is not Workrave and will not be. The same like Workrave is not Stretchly and never be. 

And as the maintainer of Stretchly, I can say that I don't see any reason why would I look at other apps. That would be the waste of my time that I don't have enough already :)
